Meme/Meme/MemeTemplatesViewModel.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  MemeTemplatesViewModel.swift
+//  Meme
+//
+//  Created by Mesfin Bekele Mekonnen on 7/30/19.
+//  Copyright © 2019 Mesfin Bekele Mekonnen. All rights reserved.
+//
+
+import Foundation
+
+protocol TemplatesViewModel {
+    func requestTemplates(completion: @escaping ([Template]) -> Void?)
```

Comment:
maybe rename to `requestMemeTemplates(completion: @escaping ([MemeTemplate]) -> Void)`
and drop the function below

Reply:
I haven't defined a `MemeTemplate` struct as of yet though, could you explain how it would be different from a Meme and a Template?